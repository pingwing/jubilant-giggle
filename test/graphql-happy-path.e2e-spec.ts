import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import * as request from 'supertest';
import { App } from 'supertest/types';
import { AppModule } from './../src/app.module';
import {
  createCharacterMutationQueryString,
  getAllCharactersQueryString,
  GraphqlResponse,
} from './graphql-documents';

describe('AppController (e2e)', () => {
  let app: INestApplication<App>;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  it('should get all characters', async () => {
    const response = await request(app.getHttpServer())
      .post('/graphql')
      .send({ query: getAllCharactersQueryString })
      .expect(200);
    const responseBody = response.body as GraphqlResponse;

    expect(responseBody?.errors).toBeUndefined();
    expect(responseBody?.data?.characters).toMatchSnapshot();
  });

  it('Happy Path 1: should create a new character', async () => {
    const response1 = await request(app.getHttpServer())
      .post('/graphql')
      .send({ query: getAllCharactersQueryString })
      .expect(200);
    const responseBody1 = response1.body as GraphqlResponse;

    expect(responseBody1?.errors).toBeUndefined();
    expect(responseBody1?.data?.characters).toHaveLength(1);

    const response2 = await request(app.getHttpServer())
      .post('/graphql')
      .send({ query: createCharacterMutationQueryString })
      .expect(200);
    const responseBody2 = response2.body as GraphqlResponse;

    expect(responseBody2?.errors).toBeUndefined();

    const response3 = await request(app.getHttpServer())
      .post('/graphql')
      .send({ query: getAllCharactersQueryString })
      .expect(200);
    const responseBody3 = response3.body as GraphqlResponse;

    expect(responseBody3?.errors).toBeUndefined();
    expect(responseBody3?.data?.characters).toHaveLength(2);


    // expect(responseBody?.data?.characters).toMatchSnapshot();
  });
});
