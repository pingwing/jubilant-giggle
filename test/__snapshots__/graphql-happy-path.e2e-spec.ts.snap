// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) should get all characters 1`] = `
[
  {
    "episodes": [
      {
        "name": "NEWHOPE",
      },
      {
        "name": "EMPIRE",
      },
      {
        "name": "JEDI",
      },
    ],
    "id": "019645d4-8273-7758-92d7-f20f20d723be",
    "name": "Luke Skywalker",
    "planet": null,
  },
  {
    "episodes": [
      {
        "name": "NEWHOPE",
      },
      {
        "name": "EMPIRE",
      },
      {
        "name": "JEDI",
      },
    ],
    "id": "01965aac-6388-765c-b3c9-ed694de57013",
    "name": "Darth Vader",
    "planet": null,
  },
  {
    "episodes": [
      {
        "name": "NEWHOPE",
      },
      {
        "name": "EMPIRE",
      },
      {
        "name": "JEDI",
      },
    ],
    "id": "01965aac-6389-7102-aaa1-8da5708abfff",
    "name": "Han Solo",
    "planet": null,
  },
  {
    "episodes": [
      {
        "name": "NEWHOPE",
      },
      {
        "name": "EMPIRE",
      },
      {
        "name": "JEDI",
      },
    ],
    "id": "01965aac-6389-7102-aaa1-90d5f5ef1320",
    "name": "Leia Organa",
    "planet": {
      "name": "Alderaan",
    },
  },
  {
    "episodes": [
      {
        "name": "NEWHOPE",
      },
    ],
    "id": "01965aac-6389-7102-aaa1-953cf11d64af",
    "name": "Wilhuff Tarkin",
    "planet": null,
  },
  {
    "episodes": [
      {
        "name": "NEWHOPE",
      },
      {
        "name": "EMPIRE",
      },
      {
        "name": "JEDI",
      },
    ],
    "id": "01965aac-6389-7102-aaa1-9b9a12cc675c",
    "name": "C-3PO",
    "planet": null,
  },
  {
    "episodes": [
      {
        "name": "NEWHOPE",
      },
      {
        "name": "EMPIRE",
      },
      {
        "name": "JEDI",
      },
    ],
    "id": "01965aac-6389-7102-aaa1-9c6272e2577d",
    "name": "R2-D2",
    "planet": null,
  },
]
`;
